operators/src/source/gdal_source.rs

```diff
@@ -370,6 +370,84 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
     }
 }
 
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataFixedTimes {
+    pub time_steps: Vec<TimeInterval>,
+    pub minimum_step: TimeStep,
+    pub time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    pub params: GdalDatasetParameters,
+    pub result_descriptor: RasterResultDescriptor,
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let valid_duration = TimeInterval::new(
+            self.time_steps[0].start(),
+            self.time_steps[self.time_steps.len() - 1].start(),
+        )?;
+
+        let parts = match valid_duration.intersect(&query.time_interval) {
+            Some(time_interval) => {
+                let mut loading_info_parts: Vec<GdalLoadingInfoPart> = Vec::new();
+
+                for (index, time_position) in self.time_steps.iter().enumerate() {
```

Comment:
the `time_position` is now an interval, so maybe call it `loading_info_validity`? Maybe you have a better idea for a name

Reply:
I couldn't come up with a better name for now, perhaps we can change it later.